File: Frontend/src/contracts/responses/planes/managementPlaneResponse.ts
// TypeScript-тип managementPlaneResponse описывает форму данных, которые frontend получает от backend.

export interface ManagementPlaneResponse {
    id: number;
    modelName: string;
    maxDistance: number;
    passengerCapacity: number;
    cruisingSpeed: number;
    flightHourCost: number;
    imageBase64?: string | null;
}

export interface SavePlaneRequest {
    modelName: string;
    maxDistance: number;
    passengerCapacity: number;
    cruisingSpeed: number;
    flightHourCost: number;
    imageBase64?: string | null;
}

File: AirCharter.API/AirCharter.API/Responses/Planes/ManagementPlaneResponse.cs
namespace AirCharter.API.Responses.Planes;

public sealed class ManagementPlaneResponse
{
    public int Id { get; set; }

    public string ModelName { get; set; } = string.Empty;

    public int MaxDistance { get; set; }

    public int PassengerCapacity { get; set; }

    public int CruisingSpeed { get; set; }

    public decimal FlightHourCost { get; set; }

    public string? ImageBase64 { get; set; }
}
